spot_driver/launch/spot_depth_publishers.launch.py

```diff
@@ -0,0 +1,80 @@
+# Copyright [2023] Boston Dynamics AI Institute, Inc.
+
+
+import launch
+import launch_ros
+import yaml
+from launch.actions import DeclareLaunchArgument, OpaqueFunction
+from launch.substitutions import LaunchConfiguration, PathJoinSubstitution
+
+
+def launch_depth_register_nodelets(
+    context: launch.LaunchContext,
+    spot_name: LaunchConfiguration,
+    camera_sources_yaml: LaunchConfiguration,
+    ld: launch.LaunchDescription,
+) -> None:
+    composable_node_descriptions = []
+
+    camera_sources_yaml = camera_sources_yaml.perform(context)
+    if not camera_sources_yaml or camera_sources_yaml == "None":
+        camera_sources = ["frontleft", "frontright", "left", "right", "back", "hand"]
+    else:
+        with open(camera_sources_yaml, "r") as yaml_file:
+            camera_sources = yaml.safe_load(yaml_file)["sources"]
```

Comment:
From the perspective of feature parity with the existing driver I think it's not actually necessary to allow users to specify which exact cameras should have registered depth image nodelets.

Within the existing spot_ros2 driver, there are essentially two factors which determine which registered depth images are published:
- The `"publish_depth_registered"` parameter: If true, request and publish registered depth images for all Spot cameras. If false, no registered depth images are published.
- Whether the Spot has an arm or not: If it does, publish all requested image types for the hand camera. If it does not, no hand camera images are published. (Note that while the driver launch files do have a "has_arm" argument, within the driver node it determines if the Spot has an arm by actually querying the robot through the SDK.)

This means that currently the option to publish registered depth images (or images of any type, for that matter) is an "all or nothing" sort of thing. This is the case [in the standalone image publisher node](https://github.com/bdaiinstitute/spot_ros2/blob/c945b72b21898284015409261c1095786471f076/spot_driver/spot_driver/publish_cameras.py#L129-L131) as well. I think it would be OK to just always create a nodelet for every expected Spot camera, and add a launch configuration argument to set whether or not we expect the hand camera to be available as well. This would simplify the configuration of your launch file quite a bit, since you wouldn't need to handle parsing the camera sources as an input YAML file.

Reply:
Noted, I'll make that change